From 47644fdd87b5a1df546551f9420839c648df48ec Mon Sep 17 00:00:00 2001
From: Sidhanta Sahu <quic_sidhanta@quicinc.com>
Date: Wed, 8 Nov 2023 09:31:16 -0800
Subject: [PATCH] wifi: ath12k: Print log before coredump

win-triage automation requires below print just before coredump.
Add the log for all type of chipsets.
	"WLAN target is restarting"

Signed-off-by: Sidhanta Sahu <quic_sidhanta@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.c | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/core.c b/drivers/net/wireless/ath/ath12k/core.c
index 58315ff..2295b90 100644
--- a/drivers/net/wireless/ath/ath12k/core.c
+++ b/drivers/net/wireless/ath/ath12k/core.c
@@ -3034,10 +3034,13 @@ static int ath12k_core_rproc_start(struct rproc *rproc)
 	return ret;
 }
 
-static void ath12k_core_rproc_coredump(struct rproc *rproc)
+static void ath12k_core_rproc_coredump(struct ath12k_base *ab,
+				       struct rproc *rproc)
 {
-	if (rproc)
+	if (rproc) {
+		ath12k_info(ab, "WLAN target is restarting");
 		rproc->ops->coredump(rproc);
+	}
 }
 
 /* Asserted target's reboot handling for crash type ATH12K_RPROC_ROOTPD_AHB_CRASH */
@@ -3062,11 +3065,11 @@ static void ath12k_core_rproc_boot_recovery(struct ath12k_base *ab)
 	ath12k_core_rproc_stop(ab_ahb->tgt_rproc);
 
 	ab_ahb->tgt_rproc->state = RPROC_SUSPENDED;
-	ath12k_core_rproc_coredump(ab_ahb->tgt_rproc);
+	ath12k_core_rproc_coredump(ab, ab_ahb->tgt_rproc);
 
 	ab_ahb->tgt_rrproc->state = RPROC_RUNNING;
 	ath12k_core_rproc_stop(ab_ahb->tgt_rrproc);
-	ath12k_core_rproc_coredump(ab_ahb->tgt_rrproc);
+	ath12k_core_rproc_coredump(ab, ab_ahb->tgt_rrproc);
 
 	ath12k_qmi_free_target_mem_chunk(ab);
 
@@ -3092,7 +3095,7 @@ static void ath12k_core_upd_rproc_boot_recovery(struct ath12k_base *ab)
 	 * Start root pd
 	 */
 	ath12k_core_rproc_stop(ab_ahb->tgt_rproc);
-	ath12k_core_rproc_coredump(ab_ahb->tgt_rproc);
+	ath12k_core_rproc_coredump(ab, ab_ahb->tgt_rproc);
 	ath12k_qmi_free_target_mem_chunk(ab);
 	ath12k_core_rproc_start(ab_ahb->tgt_rproc);
 	ag->crash_type = ATH12K_NO_CRASH;
@@ -3164,7 +3167,7 @@ static void ath12k_core_rpd_rproc_boot_recovery(struct ath12k_base *ab)
 			ath12k_err(ab, "failed to stop root_pd:%s ret:%d\n",
 				   rootpd_rproc->name, ret);
 
-		ath12k_core_rproc_coredump(rootpd_rproc);
+		ath12k_core_rproc_coredump(ab, rootpd_rproc);
 	}
 
 	ath12k_qmi_free_target_mem_chunk(ab);
-- 
2.17.1

